Sanity/pkcs11/basic/runtest.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   runtest.sh of /CoreOS/clevis/Sanity/basic/pkcs11
+#   Description: tests the basic pkcs11 functionality of clevis
+#   Author: Martin Litwora <mlitwora@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright (c) 2024 Red Hat, Inc.
+#
+#   This copyrighted material is made available to anyone wishing
+#   to use, modify, copy, or redistribute it subject to the terms
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Include Beaker environment
+. /usr/bin/rhts-environment.sh || :
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+# Include utils library containing critical functions
+lib_path="${PWD%$TMT_TEST_NAME}/lib/utils.sh"
+. $lib_path || exit 1
+
+
+PACKAGE="clevis"
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlAssertRpm $PACKAGE
+        rlRun "TMPDIR=\$(mktemp -d)" 0 "Creating tmp directory"
+        rlRun "pushd $TMPDIR"
+
+        rlRun "packageVersion=$(rpm -q ${PACKAGE} --qf '%{name}-%{version}-%{release}\n')"
+        # TODO: add correct version that will have the pkcs11 feature implemented
+        rlTestVersion "${packageVersion}" '>=' 'clevis-20-1'
+
+        install_softhsm
+        rlLog "Creating a softhsm configuration file"
+        create_hsm_config
+
+        export SOFTHSM2_CONF=$TMPDIR/softhsm.conf
+        TOKEN_LABEL="test_token"
+        SOFTHSM_LIB="/usr/lib64/softhsm/libsofthsm.so"
+        PINVALUE=1234
+        ID="0001"
+
+        rlRun -l "softhsm2-util --init-token --label $TOKEN_LABEL --free --pin $PINVALUE --so-pin $PINVALUE" 0 "Initialize token"
+        rlRun -l "pkcs11-tool --keypairgen --key-type="rsa:2048" --login --pin=$PINVALUE --module=$SOFTHSM_LIB --label=$TOKEN_LABEL --id=$ID" 0 "Generating a new key pair"
+
+        # Get serial number of the token
+        TOKEN_SERIAL_NUM=$(pkcs11-tool --module $SOFTHSM_LIB -L | grep "serial num" | awk '{print $4}')
+        URI="{\"uri\": \"pkcs11:model=SoftHSM%20v2;manufacturer=SoftHSM%20project;serial=$TOKEN_SERIAL_NUM;token=$TOKEN_LABEL;id=$ID;module-path=$SOFTHSM_LIB?pin-value=$PINVALUE\", \"mechanism\": \"RSA-PKCS\"}"
```

Comment:
I would propose to add also some test case with simpler URI (just `pkcs11:` should allow the clevis to find the only key in the system) and interactive PIN prompt with some expect script.

And a TODO for the mechanism to try also OAEP when it will be available in some test environment.

Reply:
I've added both the test cases under the https://github.com/RedHat-SP-Security/clevis-tests/pull/8/commits/294bc86f947ca1d4831e062270d5c38d5bd535a0. Both the empty `pkcs11:` uri and OAEP does not work at the moment, though.

Reply:
Thanks!.

@sarroutbi what do you think about the pin prompting. Is it implemented or in plans so far?